examples/minecraft/src/VRPlayerControl.jsx

```diff
@@ -0,0 +1,52 @@
+import { useEffect } from 'react'
+import { useFrame } from '@react-three/fiber'
+import { useXRControllerState, XROrigin } from '@react-three/xr'
+
+export function VRPlayerControl({ playerJump, playerMove }) {
+  const
+    controller_left = useXRControllerState('left'),
+    controller_right = useXRControllerState('right')
+
+  let gamepad_info = {
+    connected_right: false,
+    connected_left: false,
+    thumbstick: null,
+    a_button: null
+  }
+
+  useEffect(() => {
```

Comment:
the evaluations made in this use effect better fit directly into the use frame imo, seems overlay complex

Reply:
jump check placed directly in useFrame. useEffect removed.